core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -293,6 +294,8 @@ public void commit() {
     } catch (RuntimeException e) {
       LOG.warn("Failed to load committed table metadata, skipping manifest clean-up", e);
     }
+
+    Listeners.notifyAll(updateEvent());
```

Comment:
This needs to be in a `try`/`catch` block to ensure that no exceptions are thrown. It is important for correctness that this returns successfully if the commit was successful. If the commit succeeds, but this throws an exception, then higher-level retries will re-attempt the operation on the same data and produce duplicates.

Also, if this is sent after commit, then it should include the sequence number that was committed.

Reply:
Makes sense